### DISCUSSION TITLE
default not implemented exception for abstract methods

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#5881] [General]
```C#
public abstract class MyClass
{
    protected abstract MyMethod() => throw new Exception(); //CS0500, the desire effect would be an implicit exception
}
```
Should at least be possible, so I don't have to implement the abstract method at all if I don't want to in my derived class that isn't abstract.

**Comment by bartonjs** [comment]
Language suggestions/discussions are more appropriate at https://github.com/dotnet/csharplang/.

**Comment by huoyaoyuan** [comment]
This turns a compile time error into runtime exception, which is error prone.
The IDE already has support to fill all abstract methods with `NotImplementedException` body.

**Reply by Xyncgas** [reply]
the current IDE's way of filling all abstract methods with NotImplementedException body still takes up space on my screen.
however if in the language we say abstract methods are automatically throwing we solve the problem

**Reply by CyrusNajmabadi** [reply]
> however if in the language we say abstract methods are automatically throwing we solve the problem

That woudl be bad, because now you might miss overriding in a derived type.  If you don't want to require overriding, then just make it virtual and specify the behavior you want.

**Reply by Xyncgas** [reply]
oh I will give it to you

**Reply by huoyaoyuan** [reply]
> are automatically

I tend to never use the word "automatic" when dealing about programming language. We are the persons responsible for automation. There can be indicators as shortcuts for behavior, but implicit behavior is annoying.

**Comment by jscarle** [comment]
I agree. Virtual methods make this a moot point.
